src/api/Controllers/LayerController.cs

```diff
@@ -0,0 +1,99 @@
+﻿using BDMS.Models;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace BDMS.Controllers;
+
+[ApiController]
+[Route("api/v{version:apiVersion}/[controller]")]
+public class LayerController : ControllerBase
+{
+    private readonly BdmsContext context;
+    private readonly ILogger logger;
+
+    public LayerController(BdmsContext context, ILogger<LayerController> logger)
+    {
+        this.context = context;
+        this.logger = logger;
+    }
+
+    /// <summary>
+    /// Asynchronously gets the <see cref="Layer"/>s, optionally filtered by <paramref name="profileId"/>.
+    /// </summary>
+    /// <param name="profileId">The id of the stratigraphy containing the layers to get.</param>
+    [HttpGet]
+    public async Task<ActionResult<IEnumerable<Layer>>> GetAsync(int? profileId = null)
+    {
+        var layers = context.Layers.AsQueryable();
+
+        if (profileId != null)
+        {
+            layers = layers.Where(l => l.StratigraphyId == profileId);
+
+            if (layers.Any())
+            {
+                return await layers.AsNoTracking().ToListAsync().ConfigureAwait(false);
+            }
+            else
+            {
+                return NotFound();
+            }
+        }
+
+        return await layers.AsNoTracking().ToListAsync().ConfigureAwait(false);
+    }
+
+    /// <summary>
+    /// Asynchronously gets the <see cref="Layer"/>s, optionally filtered by <paramref name="id"/>.
+    /// </summary>
+    /// <param name="id">The id of the stratigraphy containing the layers to get.</param>
+    [HttpGet("{id}")]
+    public async Task<ActionResult<IEnumerable<Layer>>> GetByIdAsync(int id)
+    {
+        var layer = await context.Layers.SingleOrDefaultAsync(l => l.Id == id).ConfigureAwait(false);
+
+        if (layer != null)
+        {
+            return Ok(layer);
+        }
+        else
+        {
+            return NotFound();
+        }
+    }
+
+    /// <summary>
+    /// Asynchronously updates the <see cref="Layer"/> corresponding to
+    /// the <paramref name="layer"/> with the values to update.
+    /// </summary>
+    /// <param name="layer"> The <see cref="Layer"/> to update.</param>
```

Comment:
PP: Unnötigen Whitespace entfernen.
```suggestion
    /// <param name="layer">The <see cref="Layer"/> to update.</param>
```